src/lib/pickles/wrap.ml

```diff
@@ -283,518 +283,388 @@ let deferred_values (type n) ~(sgs : (Backend.Tick.Curve.Affine.t, n) Vector.t)
+           Kimchi_types.prover_proof
+
+    module type SETUP = sig
+      val example : example
+
+      (* Feature flags tused for backend proof *)
+      val actual_feature_flags : bool Plonk_types.Features.t
+    end
+
+    (* [Make] is the test functor.
+
+       Given a test setup, compute different test configurations and define 3
+       test for said configurations. *)
+    module Make (S : SETUP) = struct
+      (* Generate foreign field multiplication test backend proof using Kimchi,
+         obtaining the proof and corresponding prover index.
+
+         Note: we only want to pay the cost of generating this proof once and
+         then reuse it many times for the different recursive proof tests. *)
+      let index, public_input, proof = S.example srs
+
+      (* Obtain verifier key from prover index and convert backend proof to
+         snarky proof *)
+      let vk = Kimchi_bindings.Protocol.VerifierIndex.Fp.create index
+
+      let proof = Backend.Tick.Proof.of_backend proof
+
+      let test_feature_flags_configs =
+        generate_test_feature_flag_configs S.actual_feature_flags
+
+      let runtest feature_flags =
+        run_recursive_proof_test S.actual_feature_flags feature_flags
+          public_input vk proof
+
+      let%test "true -> yes" = runtest test_feature_flags_configs.true_is_yes
+
+      let%test "true -> maybe" =
+        runtest test_feature_flags_configs.true_is_maybe
+
+      let%test "all maybes" = runtest test_feature_flags_configs.all_maybes
+    end
+
+    (* Small combinators to lift gate example signatures to the expected
+       signatures for the tests. This amounts to generating the list of public
+       inputs from either no public inputs, a single one or a pair of inputs
+       returned by the gate example. *)
+
+    let no_public_input gate_example srs =
+      let index, proof = gate_example srs in
+      (index, [], proof)
+
+    let public_input_1 gate_example srs =
+      let index, public_input, proof = gate_example srs in
+      (index, [ public_input ], proof)
+
+    let public_input_2 gate_example srs =
+      let index, (public_input1, public_input2), proof = gate_example srs in
+      (index, [ public_input1; public_input2 ], proof)
+
+    let%test_module "lookup" =
+      ( module Make (struct
+        let example =
+          public_input_1 (fun srs ->
+              Kimchi_bindings.Protocol.Proof.Fp.example_with_lookup srs true )
+
+        let actual_feature_flags =
+          { Plonk_types.Features.chacha = false
+          ; range_check = false
+          ; foreign_field_add = false
+          ; foreign_field_mul = false
+          ; xor = false
+          ; rot = false
+          ; lookup = true
+          ; runtime_tables = true
+          }
+      end) )
+
+    let%test_module "foreign field multiplication" =
+      ( module Make (struct
+        let example =
+          no_public_input
+            Kimchi_bindings.Protocol.Proof.Fp.example_with_foreign_field_mul
```

Comment:
Maybe it's function currying..., but is there supposed to be no `srs` argument passed here, where as in the call above there is.